document-store/src/main/java/org/hypertrace/core/documentstore/postgres/PostgresCollection.java

```diff
@@ -378,6 +378,14 @@ public CloseableIterator<Document> aggregate(
     return executeQueryV1(query);
   }
 
+  @Override
+  public Optional<Document> atomicReadAndUpdateDocument(
+      final org.hypertrace.core.documentstore.query.Query query, final Document updateDocument)
+      throws IOException {
+    // TODO: Implement
+    throw new UnsupportedOperationException(UNSUPPORTED_QUERY_OPERATION);
```

Comment:
Let's have a basic implementation of it on Postgres's side too. we need parity before the API is getting used.